prime-router/src/main/kotlin/azure/DatabaseAccess.kt

```diff
@@ -93,9 +93,19 @@ class DatabaseAccess(private val create: DSLContext) : Logging {
             ?: error("Could not find $reportId that matches a task")
     }
 
-    /** Fetch multiple task records and lock them so other connections can not grab them */
-    fun fetchAndLockTasksForOneReceiver(
-        nextAction: TaskAction,
+    /**
+     * Fetch multiple task records and lock them so other connections cannot grab them.
+     * If the [at] time is not null, do an exact match against next_action_at. TODO remove this option - not used.
+     * If the [at] time is null, limit query to when next_action_at is after [backstopTime].
+     *
+     * BatchFunction's job in life is operating on a timer, so it only operates on Tasks with nonnull next_action_at.
+     *
+     * Note that this uses TASK.next_action_at as the backstopTime comparison, rather than TASK.created_at.
+     * This allows us to manually recover from failures by simply setting a TASK's next_action_at in the future.
+     * This also avoids "losing" batch tasks if a receiver's [Receiver.Timing.numberPerDay] changes while there is
+     * work-in-progress waiting to be batch.  For example, we might change the setting from once a day to once an hour.
+     */
```

Comment:
The backstop time is compared to the next_action_at time, not the created_at time.   I thought about this a lot -- the two good reasons for doing it this way are in the comment here - easier to do manual recover/retry, and it elegantly avoids the problem of "losing" batches if someone suddenly changes a receiver's timing while there's WIP.